feat(terminal): set exit code in v:event.status (#15406)
neovim/neovim: runtime/doc/nvim_terminal_emulator.txt

  programs can set this by emitting an escape sequence.
- |'channel'|  Terminal PTY |job-id|.  Can be used with |chansend()| to send
  input to the terminal.
- The |TermClose| event gives the terminal job exit code in the |v:event|
  "status" field. For example, this autocmd closes terminal buffers if the job
  exited without error: >

>     autocmd TermClose * if !v:event.status | exe 'bdelete! '..expand('<abuf>') | endif

Author: This should be just one `.` for concatenation, not two. And in fact you don’t need to concatenate the strings at all since `exe` automatically separates its arguments with a space.

Reviewer: > This should be just one `.` for concatenation, not two. And in fact you don’t need to concatenate the strings at all since `exe` automatically separates its arguments with a space.

Two is supported now and is less ambiguous. `:help expr-..`

I thought about omitting the operator but felt like it's a bit too magical and could lead to mistakes if users edit this in their config without knowing all the fun details of vimscript.

Reviewer: > I’m still surprised this is necessary. Using just `bdelete` works fine for me even with multiple terminal buffers open

Try `:terminal sleep 3` then switch to another buffer. 

Neither terminal_close() nor apply_autocmds_group() switch the current buffer, afair.

Author: > Two is supported now and is less ambiguous. `:help expr-..`

Ah that’s right, I had forgotten about that since I so rarely see it in practice. That is nice that the Lua and VimL concatenation operator can be the same.

  programs can set this by emitting an escape sequence.
- |'channel'|  Terminal PTY |job-id|.  Can be used with |chansend()| to send
  input to the terminal.
- *b:term_error*  Return code of the job that finished running in the terminal.
  Analagous to |v:shell_error|.  Use this to automatically close terminal
  buffers that finished without an error: >

>     autocmd TermClose * if !get(b:, 'term_error') | bdelete | endif

Author: Note: this particular example (using `:bdelete` without a `!`) depends on #15402 being merged.

  programs can set this by emitting an escape sequence.
- |'channel'|  Terminal PTY |job-id|.  Can be used with |chansend()| to send
  input to the terminal.
- On |TermClose|, the exit code of the terminal job is given by the "status"
  field of |v:event|. Example: >

>     autocmd TermClose * if !v:event.status | bdelete | endif

Reviewer: ```suggestion
    autocmd TermClose * if !v:event.status | exe 'bdelete! '..expand('<abuf>') | endif
```

> - The exit code of the job in a terminal buffer can be checked in the
>   |TermClose| autocmd with the "status" field of |v:event|. Example: >
>     autocmd TermClose * if !v:event.status | bdelete | endif

Reviewer: ```suggestion
- On |TermClose|, the exit code of the terminal job is given by the "status"
  field of |v:event|. Example: >
    autocmd TermClose * if !v:event.status | bdelete | endif
```
